core/net/src/api.rs

```diff
@@ -31,8 +31,52 @@ impl TryRemoteEndpoint for &str {
     }
 }
 
-pub fn net_node_id(node_id: &NodeId) -> String {
-    format!("{}/{:?}", net::BUS_ID, node_id)
+pub struct NetSrc {
+    identity: NodeId,
+}
+
+impl NetSrc {
+    pub fn to(&self, dst: NodeId) -> NetDst {
+        NetDst {
+            identity: self.identity,
+            dst,
+        }
+    }
+}
+
+pub struct NetDst {
+    identity: NodeId,
```

Comment:
Both fields are identities. I suggest to rename `identity` to `src`.

Reply:
i was thinking about `src` when i wrote this. I was thinking about identity selection for the caller.